src/test/fuzz/message.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <key_io.h>
+#include <optional.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/message.h>
+#include <util/strencodings.h>
+
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <vector>
+
+void initialize()
+{
+    static const ECCVerifyHandle ecc_verify_handle;
+    ECC_Start();
+    SelectParams(CBaseChainParams::REGTEST);
+}
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    const std::string random_message = fuzzed_data_provider.ConsumeRandomLengthString(1024);
+    {
+        const std::vector<uint8_t> random_bytes = ConsumeRandomLengthByteVector(fuzzed_data_provider);
+        CKey private_key;
+        private_key.Set(random_bytes.begin(), random_bytes.end(), fuzzed_data_provider.ConsumeBool());
```

Comment:
`ConsumeRandomLengthByteVector()` will return a result with a length between 0 and 4096. The private key will only be valid if the length is 32.

Is it the case that we will test with a valid private key in 1 of 4097 cases?

If yes, then does it make sense to add one more iteration with exactly 32 bytes, so that we (likely) end up with a valid private key?

Reply:
A modern coverage-guided fuzzer like `libFuzzer` will quickly figure out which lengths that are meaningful to test in the sense that they increase coverage, so luckily no need to hard code anything :)